File: labArray/Banco.java
package labArray;

import java.util.ArrayList;

public class Banco {
    private ArrayList<Cliente> clientes;
    private int numeroDeClientes;

    public Banco()
    {
        clientes = new ArrayList<>();
    }
    public void adicionarCliente(Cliente cliente)
    {
        clientes.add(cliente);
        numeroDeClientes++;
    }
    public Cliente getClientes(int index) {
        return clientes.get(index);
    }
    public int getNumeroDeClientes() {
        return numeroDeClientes;
    }
}

File: javascript/Banco.js
module.exports = class Banco {
  constructor() {
    if (!Banco.instance) {
      this.clientes = [];
      Banco.instance = this;
    }
    return Banco.instance;
  }
  adicionarCliente(cliente) {
    this.clientes.push(cliente);
  }
  getCliente(indice) {
    return this.clientes[indice];
  }
  getNumeroDeClientes() {
    return this.clientes.length;
  }
  getClientes(nome, sobrenome) {
    var resultado = [];
    this.clientes.forEach((element) => {
      if (element.nome === nome && element.sobrenome === sobrenome) {
        resultado.push(cliente);
      }
    });
    return resultado;
  }
};
